components/power_agent/actuator.py

```
@@ -0,0 +1,962 @@
+            nvml_idx = self._nvml_index_by_uuid[uuid]
+        except KeyError:
+            # Should not happen — _ensure_identity_map raises loudly
+            # on missing UUIDs at build time. If we get here, a GPU
+            # disappeared from NVML's view between identity-map build
+            # and now (driver crash, hot-unplug). Surface it instead
+            # of silently mis-routing.
+            raise RuntimeError(
+                f"DcgmActuator.list_running_pids: GPU UUID {uuid!r} "
+                f"(DCGM gpu_idx={gpu_idx}) is no longer visible to "
+                "NVML. Suspect mid-reconcile device hot-unplug."
+            )
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)